### ISSUE TITLE
Pytorch version?

### ISSUE THREAD

**Issue by ChristianEschen** [phymhan/supervised-gan#1]
Hi 
Which version of pytorch is used for this project?

It is not possible for me to train your DSGAN.
I have changed the inner for-loop in train.py to: for i, data in enumerate(dataset.dataset)

When I try to tran the model I get the error:
Expected 4-dimensional input for 4-dimensional weight [32, 1, 4, 4], but got input of size [3, 1, 512] instead

**Comment by phymhan** [comment]
The PyTorch version is 0.4.1. Which code were you running?

**Comment by ChristianEschen** [comment]
I finally made your code working. 
Steps:
Running python 2.7 with torch 0.41
In order to make visdom working I needed to replace the util/visualizer.py with the visualizer.py from https://github.com/junyanz/pytorch-CycleGAN-and-pix2pix/tree/master/util

Finally, I had to replace norm_layer = functools.partial(nn.InstanceNorm2d, affine=False)  the  with norm_layer = functools.partial(nn.InstanceNorm2d, affine=False, track_running_stats=True)

I have not tested if itworks with python3

**Comment by phymhan** [comment]
sorry to hear the difficulties you experienced when running the code.
however, the visualizer code was inherited directly from cyclegan-pix2pix
code. i primarily use python3 so i think it would work for python3 also.

**Comment by ChristianEschen** [comment]
Is it possible to train the model with with one class label image and input rgb image?
I guessI should use --input_nc 1 and output_nc 3. ?

And if I would like train without pretrained weights. I guess I should run not give any arguments for --pretrained_model_dir. However this gives met the error.:
./checkpoints/dsgan_model/seq_net_G1.pth

Hope you can help me out

**Comment by phymhan** [comment]
set --input_nc 1 and --output_nc 3 should work, you might also opt
out --sequential_train if train without pretrained G1 weights
